macros/src/main/scala/shine/macros/Primitive.scala

```diff
@@ -0,0 +1,416 @@
+package shine.macros
+
+import scala.annotation.{StaticAnnotation, compileTimeOnly}
+import scala.reflect.macros.blackbox
+import scala.language.experimental.macros
+
+object Primitive {
+
+  @compileTimeOnly("ExpPrimitive macro")
+  class expPrimitive extends StaticAnnotation {
+    def macroTransform(annottees: Any*): Any = macro Impl.expPrimitive
+  }
+
+  @compileTimeOnly("AccPrimitive macro")
+  class accPrimitive extends StaticAnnotation {
+    def macroTransform(annottees: Any*): Any = macro Impl.accPrimitive
+  }
+
+  @compileTimeOnly("CommandPrimitive macro")
+  class comPrimitive extends StaticAnnotation {
+    def macroTransform(annottees: Any*): Any = macro Impl.comPrimitive
+  }
+
+  class Impl(val c: blackbox.Context) {
+    import c.universe._
+
+    def expPrimitive(annottees: c.Expr[Any]*): c.Expr[Any] = {
+      annottees.map(_.tree) match {
+        case (cdef: ClassDef) :: Nil =>
+          c.Expr(expPrimitivesFromClassDef(cdef))
+        case (cdef: ClassDef) :: (md: ModuleDef) :: Nil =>
+          c.Expr(q"{${expPrimitivesFromClassDef(cdef)}; $md}")
+        case _ => c.abort(c.enclosingPosition, "expected a class definition")
+      }
+    }
```

Comment:
Could these three `*Primitive` functions be made into one function that accepts the `ClassDef => ClassDef` transformations as a a parameter?